chainntnfs/neutrinonotify/neutrino.go

```diff
@@ -527,6 +481,79 @@ chainScan:
 	return false
 }
 
+// handleBlocksConnected applies a chain update for a new block. Any watched
+// transactions included this block will processed to either send notifications
+// now or after numConfirmations confs.
+func (n *NeutrinoNotifier) handleBlockConnected(newBlock *filteredBlock) error {
+	// First we'll notify any subscribed clients of the
+	// block.
+	n.notifyBlockEpochs(int32(newBlock.height), &newBlock.hash)
+
+	// Next, we'll scan over the list of relevant
+	// transactions and possibly dispatch notifications for
+	// confirmations and spends.
+	for _, tx := range newBlock.txns {
+		// Check if the inclusion of this transaction
+		// within a block by itself triggers a block
+		// confirmation threshold, if so send a
+		// notification. Otherwise, place the
+		// notification on a heap to be triggered in
+		// the future once additional confirmations are
+		// attained.
+		mtx := tx.MsgTx()
+		txIndex := tx.Index()
+		txSha := mtx.TxHash()
+		n.checkConfirmationTrigger(&txSha, newBlock, txIndex)
+
+		for i, txIn := range mtx.TxIn {
+			prevOut := txIn.PreviousOutPoint
+
+			// If this transaction indeed does
+			// spend an output which we have a
+			// registered notification for, then
+			// create a spend summary, finally
+			// sending off the details to the
+			// notification subscriber.
+			if clients, ok := n.spendNotifications[prevOut]; ok {
```

Comment:
A similar nesting optimization can be made here: we'd instead check for `!ok` above and continue if so.